vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/transformer/AbstractFetchingUrlsTableTransformer.java

```diff
@@ -109,7 +111,9 @@ public void setHttpRedirectsProvider(HttpRedirectsProvider httpRedirectsProvider
 
     protected URI getMainApplicationPageUri()
     {
-        return webApplicationConfiguration.getMainApplicationPageUrl();
+        URI mainApplicationPageUrl = webApplicationConfiguration.getMainApplicationPageUrl();
+        isTrue(mainApplicationPageUrl != null, "URL of the main application page should be non-blank");
+        return mainApplicationPageUrl;
```

Comment:
I believe there is no reason to check the value each time the method invoked, consider caching the result (`Suppliers.memoize`?)